pkg/image/graph/nodes/nodes.go

```diff
@@ -44,22 +44,13 @@ func FindImage(g osgraph.MutableUniqueGraph, imageName string) graph.Node {
 // not already exist. If the reference is invalid, the Name field of the graph will be used directly.
 func EnsureDockerRepositoryNode(g osgraph.MutableUniqueGraph, name, tag string) graph.Node {
 	ref, err := imageapi.ParseDockerImageReference(name)
-	if err == nil {
-		if len(tag) != 0 {
-			ref.Tag = tag
-		}
-		if len(ref.Tag) == 0 {
-			ref.Tag = imageapi.DefaultImageTag
-		}
-		if len(ref.Registry) == 0 {
-			ref.Registry = "docker.io"
-		}
-		if len(ref.Namespace) == 0 {
-			ref.Namespace = imageapi.DockerDefaultNamespace
-		}
-	} else {
+	if err != nil {
 		ref = imageapi.DockerImageReference{Name: name}
 	}
+	if len(tag) != 0 {
+		ref.Tag = tag
+	}
+	ref = ref.DockerClientDefaults()
```

Comment:
agree... defaulting (when intent is clear) is ok in the CLI commands or (maybe) in the API, but not in the graph